### DISCUSSION TITLE
当产生标注时，如何自动同步到本地

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2220] [服务]
# 重要提示

> ### 嗨，简悦全新的 [教程中心](https://www.yuque.com/kenshin/simpread) 已上线，**下面的内容并不一定是最新的**，建议 [访问这里](https://www.yuque.com/kenshin/simpread/wkswh7) 查看最新版教程。

***

> 稍后读一个很重要的功能：Capture。所以，当产生标注时，如何也能方便的做保存到本地处理？

> 此功能可以跟 [**如何让稍后读支持读取本地缓存**](https://github.com/Kenshin/simpread/discussions/2146) 配合。

> **[简悦 2.2.0](http://ksria.com/simpread/welcome/version_2.2.0.html)** 发布后，就可以很容易的实现这种方式，且有更强的定制化。

## 前提

> 简悦因为灵活的使用方式导致此功能并没有所谓的「开关」开启它，需要多种功能协同完成，但过程并不困难。

## 前期准备

1. 需要 [**简悦 · 高级账户**](https://simpread.pro/price.html)
2. 需要使用并配置 简悦 · 同步助手的 **[增强导出功能](http://ksria.com/simpread/docs/#/Sync?id=%e5%af%bc%e5%87%ba%e6%9c%8d%e5%8a%a1)**

### 配置 [自定义标题](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=%e8%87%aa%e5%ae%9a%e4%b9%89%e5%af%bc%e5%87%ba%e6%a0%87%e9%a2%98)

> 设置为 `{{id}}{{un_title}}{{mode}}`，如下图所示

![image](https://user-images.githubusercontent.com/81074/119208921-c4a75380-bad6-11eb-88eb-62c70d1109cc.png)

### 配置 **[增强导出](http://ksria.com/simpread/docs/#/Sync?id=%e5%af%bc%e5%87%ba%e6%9c%8d%e5%8a%a1)**

> 选项页 → 服务 → 增强导出，打开下面的选项

![image](https://user-images.githubusercontent.com/81074/119209076-64fd7800-bad7-11eb-81ee-bfe5e5da16e9.png)

### 配置 [自动化](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%8C%96)

> 加入一个新的自动化方案，简单的说：就是**当生成标注时，执行导出 HTML 的功能**，如下图所示

![image](https://user-images.githubusercontent.com/81074/119210482-df7dc600-bade-11eb-8a75-0073806ad34f.png)

> 简单的引申下，你可以在导出 HTML 时，也导出到你常见的服务。（这个话题跟本教程无关，就不在引申讨论。

### 配置结束

> 截至到现在为止，全部的配置已经结束，接下来是如何使用。

## 使用

> 需要先进入阅读模式后，加入标注后会有类似这样的提示

![2021-05-22_09-22-07 (1)](https://user-images.githubusercontent.com/81074/119210584-84000800-badf-11eb-93f0-3d722986d3f4.gif)

> 然后在 同步助手设置的 `output` 文件夹中会找到下面的文件（两个文件）

![image](https://user-images.githubusercontent.com/81074/119210594-97ab6e80-badf-11eb-9664-53320f1cc852.png)

![2021-05-22_09-26-54 (1)](https://user-images.githubusercontent.com/81074/119210673-16a0a700-bae0-11eb-9abc-ee20df62d48d.gif)

> 同样，此功能也可以在稍后读中使用

![2021-05-22_09-40-11 (1)](https://user-images.githubusercontent.com/81074/119210960-02f64000-bae2-11eb-978a-ab0a92aa5dc3.gif)

## 备注

> 上面的操作中我使用了 **HTML** 而非 **离线 HTML**（其实是一致的）

## 引申

> 此功能应该跟 [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146) 配合，同时在稍后读的阅读模式中也可以使用。

> 为什么是两个文件，请看这里 [**当产生标注时，如何同时下载标注与原文**](https://github.com/Kenshin/simpread/discussions/2147)

> 如果你是双链笔记使用者的话，还可以使用此方式 [如何配置永久链接（在双链笔记中做源链接回溯）](https://github.com/Kenshin/simpread/discussions/3188)。

## 参考知识点

> 上述操作都已经写到了文档里面，有感兴趣的用户，可以看下。

- 简悦 · 同步助手的 [增强导出功能](http://ksria.com/simpread/docs/#/Sync?id=%e5%af%bc%e5%87%ba%e6%9c%8d%e5%8a%a1)
- [自定义标题](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=%e8%87%aa%e5%ae%9a%e4%b9%89%e5%af%bc%e5%87%ba%e6%a0%87%e9%a2%98)
- [自动化](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%8C%96)

**Comment by karekin** [comment]
你好，我想问两个问题：

1. 配置自定义标题的时候，刷新之后记录没保存，之前输入的 `{{id}}{{un_title}}{{mode}}`和开关都恢复原样了
2. 标注保存到本地后，打开html文件内容只剩下标注了
![image](https://user-images.githubusercontent.com/47046461/127624394-1a9372cb-2105-4cb0-9562-698a433974c1.png)
请问我哪里弄错了吗？

**Reply by Kenshin** [reply]
> 配置自定义标题的时候，刷新之后记录没保存，之前输入的 {{id}}{{un_title}}{{mode}}和开关都恢复原样了

估计是配置被其他页面覆盖造成的。你可以这样做：

1. 设置完毕
2. 重启浏览器

> 标注保存到本地后，打开html文件内容只剩下标注了

应该是设置造成的，选项页 → 高级设定 → 标注，看下图

![image](https://user-images.githubusercontent.com/81074/127654356-e3c85208-a58b-4998-bf94-02bd59fb93db.png)

**Comment by evenz82** [comment]
你好，我导出MARKDOWN文件只有标注，没有文件。设置如下：
![image](https://user-images.githubusercontent.com/70847463/151182086-c8278970-418f-4611-8b39-495b6f4c6fe1.png)
![image](https://user-images.githubusercontent.com/70847463/151181660-8e0788bd-84bf-411d-8dd6-422d009ac24f.png)
请问还需要怎么操作呢？

**Reply by Kenshin** [reply]
这篇配置最上面提示了，还需要配置稍后读的自动化 https://github.com/Kenshin/simpread/discussions/2146

![image](https://user-images.githubusercontent.com/81074/151279748-8df80f1c-ebbb-44cc-9a2e-a6a386103612.png)
